Sources/Tuvi.Core.DataStorage.Impl/DataStorage.cs

```diff
@@ -612,6 +635,14 @@ private static void UpdateFolderUnreadCount(SQLiteConnection connection,
 
         private static void UpdateContactsUnreadCount(SQLiteConnection connection, Entities.Message newMessage, Entities.Message oldMessage = null)
         {
+            int folderId = (newMessage ?? oldMessage).FolderId;
+            var folder = connection.Find<Folder>(folderId);
+            if (folder is null || (!folder.IsInbox && !folder.IsSent))
+            {
+                // Actually, only INBOX and SENT messages are taken into account when we collect contacts
+                return;
+            }
+
             if (newMessage is null && oldMessage is null)
```

Comment:
Put this condition before line 638: int folderId = (newMessage ?? oldMessage).FolderId;

Reply:
done